lib/knapsack_pro/client/connection.rb

```diff
@@ -156,6 +155,22 @@ def get
           http.get(uri, json_headers)
         end
       end
+
+      def max_request_retries
+        # when user defined max request retries
+        return KnapsackPro::Config::Env.max_request_retries if KnapsackPro::Config::Env.max_request_retries
```

Comment:
Do you think we should limit this setting in any way? E.g. we could do:
```ruby
return [KnapsackPro::Config::Env.max_request_retries, MAX_RETRY ].min if KnapsackPro::Config::Env.max_request_retries
```

Reply:
This might not be needed because for every next retry the waiting window is increasing by 8s so they should rather not flod our API.

See example waiting time in spec
https://github.com/KnapsackPro/knapsack_pro-ruby/blob/4be8142aa3cce05fdf5a82c793f3d857f9e0e1ad/spec/knapsack_pro/client/connection_spec.rb#L175,L204

Reply:
Oh, good to know, thank you!